core/src/main/java/bisq/core/api/CorePriceService.java

```diff
@@ -65,6 +82,44 @@ public double getMarketPrice(String currencyCode) throws ExecutionException, Int
                 .collect(Collectors.toList());
     }
 
+    /** 
+     *  @return Data for market depth chart  
+    */
+    public MarketDepthInfo getMarketDepth(String currencyCode) {        
+        List<OpenOffer> buyOffers = openOfferManager.getOpenOffersByCurrency(Direction.BUY.name(), currencyCode); 
+        List<OpenOffer> sellOffers = openOfferManager.getOpenOffersByCurrency(Direction.SELL.name(), currencyCode); 
+
+        // Create buyer hashmap {key:price, value:count}
+        HashMap<Double,Long> buyHM = new HashMap<Double,Long>();        
+        buyOffers.forEach((openOffer) -> {
+            Double price = openOffer.getOffer().getOfferPayload().getCoinAmount();
```

Comment:
`getCoinAmount` result is saved as `price`, is that the correct value? But the OfferPayload has `price` only _if fixed price is used (usePercentageBasedPrice = false), otherwise 0_

Reply:
Oh interesting!  In this case getCoinAmount() is using the `amount` field from OfferPayload rather than `price` field, does that make a difference?  If not can you explain further what it means when there is no fixed price?  Is that like a market offer based on the current market price at the time @premek?

Reply:
> Is that like a market offer based on the current market price at the time

Yes that's how I understand it

Reply:
Ah ok,  I was fundamentally misunderstanding some of the variables involved.  But I see a solution now.